pkg/whip/whip_handler.go

```diff
@@ -394,12 +416,27 @@ func (h *whipHandler) addTrack(track *webrtc.TrackRemote, receiver *webrtc.RTPRe
 	}
 }
 
-func (h *whipHandler) newMediaSink(track *webrtc.TrackRemote) (MediaSink, error) {
+func (h *whipHandler) getMediaSink(track *webrtc.TrackRemote, trackQuality livekit.VideoQuality) (MediaSink, error) {
 	kind := streamKindFromCodecType(track.Kind())
 
 	if h.sdkOutput != nil {
-		// pasthrough
-		return NewSDKMediaSink(h.logger, h.params, h.sdkOutput, track, h.outputSync.AddTrack(), func() {
+		h.trackSDKMediaSinkLock.Lock()
+		defer h.trackSDKMediaSinkLock.Unlock()
+
+		if _, ok := h.trackSDKMediaSink[kind]; !ok {
```

Comment:
Just curious about why we need two add all expected simulcast tracks when we get the first one here, instead of adding each one when we get the corresponding source track callback?

Reply:
We create the tracks at signaling time. This is so we can block until they are all resolved. 

We don’t know the SSRC until the media is flowing though.

maybe a better way to do this. Don’t know yet